From a358a2ac37e86bf39c9f8a22e8e6fdffcc0f12e6 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Tue, 17 Dec 2024 18:59:17 +0200
Subject: [PATCH] [FSW-9022] make: fix for linkage of cmn lib source

Fix wrong adding of source file in embedded common,
required for linkage of get_64bit_addr_from_32bit_addr.

Change-Id: Id9093955359c9eb9a94adf705a12ec5484bd1e77
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/Makefile b/Makefile
index bc6efb9..e7f4123 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,6 @@
 PPBOOT_SRCS_C += $(shell ls app/ppboot/*.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
-PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/arch/$(HL_DEV_ARCH)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/lib_common.c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/pci.c)
 PPBOOT_APP_S := $(shell ls app/ppboot/crt0_ppboot.S)
diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index c197766..86ba7d6 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -16,6 +16,7 @@
 PPBOOT_SRCS_C := $(filter-out lib/common/module_i2c_slave.c,$(PPBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(filter-out lib/common/pif_ts.c,$(PPBOOT_SRCS_C))
 PPBOOT_SRCS_C += $(HL_DEV_ARCH)/pll.c
+PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/arch/$(BASE_DEV_NAME)/common.c)
 
 # TODO: uncomment when bootrom exports is available
 #ifeq ($(HL_TARGET),onix)
